src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1176,6 +1185,14 @@ else if (!referencePath.isDirectory())
                 // path.
                 Pattern pattern = Pattern.compile(SUBMODULE_REMOTE_PATTERN_STRING, Pattern.MULTILINE);
                 Matcher matcher = pattern.matcher(cfgOutput);
+
+                ExecutorService executorService;
```

Comment:
Naming the threads for debugging purposes in production would be a good idea.

Some useful resources
* see https://dzone.com/articles/executorservice-10-tips-and
* https://github.com/jenkinsci/jenkins/blob/jenkins-2.137/core/src/main/java/hudson/Proc.java#L141